### ISSUE TITLE
indeterminist result and different embeddings array in each execusion 

### ISSUE THREAD

**Issue by nnadine25** [IBCNServices/pyRDF2Vec#52]
hi , 
i run this code in pycharm IDE, but i get another embedding array , not the one in your example

import pandas as pd

from pyrdf2vec import RDF2VecTransformer
from pyrdf2vec.embedders import Word2Vec
from pyrdf2vec.graphs import KG
from pyrdf2vec.walkers import RandomWalker

data = pd.read_csv("samples/countries-cities/entities.tsv", sep="\t")
entities = [entity for entity in data["location"]]
print(entities)
# [
#    "http://dbpedia.org/resource/Belgium",
#    "http://dbpedia.org/resource/France",
#    "http://dbpedia.org/resource/Germany",
# ]

transformer = RDF2VecTransformer(
    Word2Vec(epochs=10),
    walkers=[RandomWalker(4, 10, with_reverse=False, n_jobs=2)],
    # verbose=1
)
embeddings, literals = transformer.fit_transform(
    KG(
        "https://dbpedia.org/sparql",
        skip_predicates={"www.w3.org/1999/02/22-rdf-syntax-ns#type"},
        literals=[
            [
                "http://dbpedia.org/ontology/wikiPageWikiLink",
                "http://www.w3.org/2004/02/skos/core#prefLabel",
            ],
            ["http://dbpedia.org/ontology/humanDevelopmentIndex"],
        ],
    ),
    entities
)
print(embeddings)
# [
#     array([ 1.5737595e-04,  1.1333118e-03, -2.9838676e-04,  ..., -5.3064007e-04,
#             4.3192197e-04,  1.4529384e-03], dtype=float32),
#     array([-5.9027621e-04,  6.1689125e-04, -1.1987977e-03,  ...,  1.1066757e-03,
#            -1.0603866e-05,  6.6087965e-04], dtype=float32),
#     array([ 7.9996325e-04,  7.2907173e-04, -1.9482171e-04,  ...,  5.6251377e-04,
#             4.1435464e-04,  1.4478950e-04], dtype=float32)
# ]

the array that i find change in each execusion : i set  the envirenement variable PYTHONHASHSEED
  and  i use random_state=42 , i use pyrdf2vec version 0.2.3

**Comment by rememberYou** [comment]
Check again. Your embeddings are similar to the ones in the example.

**Comment by nnadine25** [comment]
thanks , its writen that  Finally, to ensure random determinism for Word2Vec, you must specify a single worker:
 Word2Vec(workers=1) : where i can do this ?? i replae the number 4 with 1 in this equation ??

walkers=[RandomWalker(4, 10, with_reverse=False, n_jobs=2)],

**Comment by rememberYou** [comment]
Everything is explained in the FAQ (https://github.com/IBCNServices/pyRDF2Vec#faq). You must specify to `Word2Vec` that you use only one worker: `Word2Vec(epochs=10)` becomes `Word2Vec(workers=1, epochs=10)` . The walker and sampler have nothing to do with it. You should use the following snippet:
        
```python
import pandas as pd

from pyrdf2vec import RDF2VecTransformer
from pyrdf2vec.embedders import Word2Vec
from pyrdf2vec.graphs import KG
from pyrdf2vec.walkers import RandomWalker

data = pd.read_csv("samples/countries-cities/entities.tsv", sep="\t")
entities = [entity for entity in data["location"]]
print(entities)
# [
#    "http://dbpedia.org/resource/Belgium",
#    "http://dbpedia.org/resource/France",
#    "http://dbpedia.org/resource/Germany",
# ]

transformer = RDF2VecTransformer(
    Word2Vec(workers=1, epochs=10),
    walkers=[RandomWalker(4, 10, with_reverse=False, n_jobs=2)],
    # verbose=1
)
embeddings, literals = transformer.fit_transform(
    KG(
        "https://dbpedia.org/sparql",
        skip_predicates={"www.w3.org/1999/02/22-rdf-syntax-ns#type"},
        literals=[
            [
                "http://dbpedia.org/ontology/wikiPageWikiLink",
                "http://www.w3.org/2004/02/skos/core#prefLabel",
            ],
            ["http://dbpedia.org/ontology/humanDevelopmentIndex"],
        ],
    ),
    entities
)
print(embeddings)
# [
#     array([ 1.5737595e-04,  1.1333118e-03, -2.9838676e-04,  ..., -5.3064007e-04,
#             4.3192197e-04,  1.4529384e-03], dtype=float32),
#     array([-5.9027621e-04,  6.1689125e-04, -1.1987977e-03,  ...,  1.1066757e-03,
#            -1.0603866e-05,  6.6087965e-04], dtype=float32),
#     array([ 7.9996325e-04,  7.2907173e-04, -1.9482171e-04,  ...,  5.6251377e-04,
#             4.1435464e-04,  1.4478950e-04], dtype=float32)
# ]
```

**Comment by nnadine25** [comment]
thank you , i copy your code , and i do all what you have said , i find the same embeddings in each execusion , but the value of the array are different from your array :


[array([-0.0097891 ,  0.03753381,  0.00094535, -0.00118255,  0.02680888,
       -0.06301225,  0.0445633 ,  0.08752315, -0.04737359, -0.02216851,
       -0.00132478, -0.063329  ,  0.00883174,  0.0182234 ,  0.02503178,
       -0.03205967,  0.02488577, -0.0221401 , -0.04126531, -0.11016962,
        0.02394318,  0.02198377,  0.02858865, -0.01887381,  0.00452158,
       -0.013144  , -0.01909473, -0.01319149, -0.03265752,  0.00284413,
        0.03095006, -0.00918708,  0.05395079, -0.06663942, -0.02311148,
        0.04462403,  0.01274173, -0.05501807,  0.00918782, -0.08195817,
       -0.00777199, -0.02133708, -0.03602652, -0.02331506,  0.03149088,

**Comment by rememberYou** [comment]
This is completely normal. I did not use the same PYTHONHASHSEED as you did. Unfortunately, I have no idea what random seed was used for the example. The main thing is that you have your random determinism as desired.

**Comment by nnadine25** [comment]
i set the PYTHONHASHSEED = 42 , and random_state=42
